src/components/action-bar/action-bar-item/action-bar-item.tsx

```diff
@@ -0,0 +1,123 @@
+import { Component, Event, EventEmitter, h, Prop } from '@stencil/core';
+import { ActionBarItem } from '../action-bar.types';
+import { createRandomString } from '../../../util/random-string';
+import { ListSeparator } from 'src/interface';
+
+/**
+ * @private
+ */
+@Component({
+    tag: 'limel-action-bar-item',
+    shadow: false,
+    styleUrl: 'action-bar-item.scss',
+})
+export class ActionBarButton {
+    /**
+     * Item that is placed in the action bar.
+     */
+    @Prop()
+    public item!: ActionBarItem | ListSeparator;
+
+    /**
+     * Fired when a action bar item has been clicked.
+     */
+    @Event()
+    public select: EventEmitter<ActionBarItem | ListSeparator>;
+
+    /**
+     * Used to attach the right tooltip to the right button
+     */
+    private tooltipId: string;
+
+    constructor() {
+        this.tooltipId = createRandomString();
+    }
+
+    public render() {
+        return this.renderButton(this.handleClick);
+    }
```

Comment:
It seems pretty meaningless to break out _the whole_ render function into a single separate function…?

Reply:
I agree

Reply:
⚡